File: r_code/test_example/R/convert_distance.R
## Function is taken from https://discdown.org/rprogramming/testing.html#tinytest-in-r

#' Converting temperature in Celsius to Farenheit
#' 
#' @param C_temp
#' @returns Temperature in Farenheit
#' @examples
#' C_to_F(10)
meters2feet <- function(x) {

    if (!is.numeric(x)) {
        stop("The distance must be a number.") 
    }
    if (x < 0) {
        stop("The distance must be a non-negative number.")
    }
    return(3.28084 * x)

}  

File: python_code/test_example/convert_distance.py
from pandas.api.types import is_numeric_dtype

def meters2feet(
    x: float,       
) -> float:

    if not is_numeric_dtype(type(x)):
        raise ValueError('The distance must be a number.') 
    
    if (x < 0):
        raise ValueError('The distance must be a non-negative number.')
    
    return 3.28084 * x